### PR TITLE
Update redis gems

### PR DESCRIPTION
**Pull request by jasonpenny**
Updates our fork for some upstream changes that are required for `platform` to use a newer version of the `redis` gem

This also updates our fork repo to be able to run `bundle exec rake` locally to run all the tests

for use in `platform`:
- https://github.com/Appboy/platform/pull/76763

---

Includes (modified code from) these upstream PRs:
- https://github.com/sidekiq/sidekiq/pull/5043
- https://github.com/sidekiq/sidekiq/pull/5139

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,9 +3,9 @@ source "https://rubygems.org"
 gemspec
 
 gem "rake"
-gem "redis-namespace", github: "resque/redis-namespace", branch: :master
+gem "redis-namespace", "1.11.0"
```
Comment (jasonpenny) [inline Gemfile:6]: This is the version that `platform` is targeting to upgrade to
- https://github.com/Appboy/platform/pull/76763

File: Gemfile [hunk 2]
```diff
@@ -3,9 +3,9 @@ source "https://rubygems.org"
...
 gem "rails", ">= 6.0.2"
-gem "sqlite3", platforms: :ruby
+gem "sqlite3", "~> 1.7", platforms: :ruby
```
Comment (jasonpenny) [inline Gemfile:8]: This allows `gemtester` to work, so it seems good to do here (the Gemfile gets used for CI/tests only)

File: lib/sidekiq/client.rb [hunk 3]
```diff
@@ -186,8 +186,8 @@ def enqueue_in(interval, klass, *args)
 
     def raw_push(payloads)
       @redis_pool.with do |conn|
-        conn.multi do
-          atomic_push(conn, payloads)
+        conn.pipelined do |pipeline|
```
Comment (mattwd7) [inline lib/sidekiq/client.rb:189]: I assume this is safe to change from a `multi` to a `pipeline` b/c [the latest sidekiq does the same](https://github.com/sidekiq/sidekiq/blob/145601d4cbcab03f999ab54e75d884e7411a3bd3/lib/sidekiq/client.rb#L224-L230), but what is the reason for this?

File: lib/sidekiq/fetch.rb [hunk 4]
```diff
@@ -31,7 +31,7 @@ def initialize(options)
       @queues = @options[:queues].map { |q| "queue:#{q}" }
       if @strictly_ordered_queues
         @queues.uniq!
-        @queues << TIMEOUT
+        @queues << {timeout: TIMEOUT}
```
Comment (mattwd7) [inline lib/sidekiq/fetch.rb:34]: what is the significance of this change?

File: lib/sidekiq/client.rb (hunk 3 above)
Comment (jasonpenny) [reply lib/sidekiq/client.rb:189]: the reason is linked in this PR's description, https://github.com/sidekiq/sidekiq/pull/5043

also, some existing platform code is causing this to be a nested `multi`, which now raises an error

File: lib/sidekiq/fetch.rb (hunk 4 above)
Comment (jasonpenny) [reply lib/sidekiq/fetch.rb:34]: passing the timeout as an int is deprecated

https://github.com/redis/redis-rb/blob/0afcf1cad6f4fc74213f2b4c7a20819581c36897/lib/redis/commands/lists.rb#L253-L262

**Review by mattwd7 — APPROVED** [review]

**Pull request merged**